### PR TITLE
Introduce work-agent

### PR DESCRIPTION
**Pull request by clyang82**
1. remove `clone` sub-command.
2. introduce `agent` sub-command.
3. deploy `maestro-agent`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -270,7 +270,12 @@ ocm get /api/maestro/v1/resources
 }
 ```
 
-#### Run in CRC
+#### Run in OpenShift
+
+Take OpenShift Local as an example to deploy the maestro. If you want to deploy maestro in an OpenShift cluster, you need to set the `external_apps_domain` environment variable to point your cluster.
+```shell
+$ export external_apps_domain=apps.<company_name>.io
```
Comment (morvencao) [inline README.md:277]: nit: maybe get base domain from ingresscontroller:
```
oc -n openshift-ingress-operator get ingresscontroller default -o jsonpath='{.status.domain}'
```

**Comment by skeeey** [comment]
LGTM

File: cmd/maestro/agent/cmd.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		glog.Fatalf("error running command: %v", err)
+	}
+}
+
+func addFlags(flags *pflag.FlagSet) {
+	// workloadAgentOptions
+	flags.DurationVar(&agentOption.StatusSyncInterval, "status-sync-interval", agentOption.StatusSyncInterval, "Interval to sync resource status to hub")
+	flags.DurationVar(&agentOption.AppliedManifestWorkEvictionGracePeriod, "appliedmanifestwork-eviction-grace-period",
+		agentOption.AppliedManifestWorkEvictionGracePeriod, "Grace period for appliedmanifestwork eviction")
+	flags.StringVar(&commonOptions.SpokeClusterName, "cluster-name", commonOptions.SpokeClusterName, "Name of the cluster")
```
Comment (qiujian16) [inline cmd/maestro/agent/cmd.go:74]: we can change the flag here aligning to maestro concept. e.g. cluster-name -> consmer-id. appliedmanifestwork-eviction-grace-period" to resource-eviction-grace-period

**Comment by clyang82** [comment]
```
$ go clean -testcache
$ make test && make test-integration
CGO_ENABLED=1 GOEXPERIMENT=boringcrypto go install -ldflags="" ./cmd/maestro
OCM_ENV=testing gotestsum --format short-verbose -- -p 1 -v  \
		./pkg/... \
		./cmd/...
EMPTY pkg/api
EMPTY pkg/api/openapi
EMPTY pkg/api/presenters
EMPTY pkg/auth
EMPTY pkg/client/cloudevents
EMPTY pkg/client/ocm
PASS pkg/config.TestConfigReadStringFile (0.00s)
PASS pkg/config.TestConfigReadIntFile (0.00s)
PASS pkg/config.TestConfigReadBoolFile (0.00s)
PASS pkg/config.TestConfigReadQuotedFile (0.00s)
PASS pkg/config
PASS pkg/controllers.TestControllerFramework (0.00s)
PASS pkg/controllers
EMPTY pkg/dao
EMPTY pkg/dao/mocks
EMPTY pkg/db
EMPTY pkg/db/db_context
EMPTY pkg/db/db_session
EMPTY pkg/db/migrations
EMPTY pkg/db/mocks
EMPTY pkg/db/transaction
PASS pkg/errors.TestErrorFormatting (0.00s)
PASS pkg/errors.TestErrorFind (0.00s)
PASS pkg/errors
EMPTY pkg/handlers
EMPTY pkg/logger
PASS pkg/services.TestSQLTranslation (0.03s)
PASS pkg/services.TestResourceFindByConsumerID (0.00s)
PASS pkg/services
EMPTY pkg/util
EMPTY cmd/maestro
EMPTY cmd/maestro/agent
PASS cmd/maestro/environments.TestLoadServices (0.23s)
PASS cmd/maestro/environments
EMPTY cmd/maestro/migrate
EMPTY cmd/maestro/servecmd
EMPTY cmd/maestro/server
EMPTY cmd/maestro/server/logging

DONE 10 tests in 8.883s
CGO_ENABLED=1 GOEXPERIMENT=boringcrypto go install -ldflags="" ./cmd/maestro
OCM_ENV=testing gotestsum --format short-verbose -- -p 1 -ldflags -s -v -timeout 1h  \
			./test/integration
I1213 15:08:15.023346   29291 framework.go:76] Initializing testing environment
I1213 15:08:15.168951   29291 framework.go:161] Using Mock OCM Authz Client
I1213 15:08:15.171625   29291 framework.go:199] Disabling Sentry error reporting
{"level":"info","ts":1702451295.173663,"logger":"fallback","caller":"v2/protocol.go:124","msg":"subscribing to topics: map[sources/clusters/+/specresync:{1 0 false false} sources/maestro-integration/clusters/+/status:{1 0 false false}]"}
I1213 15:08:15.203572   29291 api_server.go:145] Serving without TLS at localhost:8000
I1213 15:08:15.203673   29291 healthcheck_server.go:56] Serving HealthCheck without TLS at localhost:8083
PASS test/integration.TestConsumerGet (0.14s)
PASS test/integration.TestConsumerPost (0.14s)
PASS test/integration.TestConsumerPaging (0.15s)
PASS test/integration.TestControllerRacing (1.31s)
PASS test/integration.TestResourceGet (0.30s)
PASS test/integration.TestResourcePost (2.17s)
PASS test/integration.TestResourcePatch (0.22s)
PASS test/integration.TestResourcePaging (0.18s)
PASS test/integration.TestResourceListSearch (0.18s)
PASS test/integration.TestUpdateResourceWithRacingRequests (0.27s)
I1213 15:08:20.277269   29291 api_server.go:151] Web server terminated
PASS test/integration

DONE 10 tests in 9.237s
```

File: cmd/maestro/agent/cmd.go (hunk 2 above)
Comment (clyang82) [reply cmd/maestro/agent/cmd.go]: Done

**Review by qiujian16 — APPROVED** [review]

**Pull request merged**